cogs/todo.py

```diff
@@ -69,38 +86,114 @@ def check(reac, user):
 
     @commands.command(name="list", aliases=["l"])
     async def _list(self, ctx):
-        l = self.bot.data.get(str(ctx.author.id))
+        todo_list = self.bot.data["todo"].get(str(ctx.author.id))
         todos = ""
-        if not l:
+        if not todo_list:
             return await ctx.send(f">>> {ctx.author.mention}のTODOリスト\n\nNone")
 
-        for i, todo in enumerate(l):
+        for i, todo in enumerate(todo_list):
             todos += f"{i}: {todo}" + "\n"
         todos = todos.rstrip("\n")
         msg = f">>> {ctx.author.mention}のTODOリスト\n\n" + todos
         return await ctx.send(msg)
 
     @commands.command(aliases=["d"])
     async def delete(self, ctx, num: int):
-        l = self.bot.data.get(str(ctx.author.id))
-        if not l:
+        todo_list = self.bot.data["todo"].get(str(ctx.author.id))
+        if not todo_list:
             return await ctx.send("> そのTODOはありません。`todo!list`で確認してください。")
         try:
-            deleted_todo = self.bot.data[str(ctx.author.id)][num]
-            del self.bot.data[str(ctx.author.id)][num]
+            deleted_todo = self.bot.data["todo"][str(ctx.author.id)][num]
+            del self.bot.data["todo"][str(ctx.author.id)][num]

```

Comment:
```suggestion
            deleted_todo = todo_list[num]
            del todo_list[num]
```
上で `todo_list` 定義してるんだし、そのまま使えば良さそう

Reply:
`todo_list`内のデータを削除すれば`self.bot.data`内のデータも一緒に削除できるんですか？

Reply:
あ、検証してみたところ消されました
Pythonの参照渡しというのはdict.getでも働くんですね...
ありがとうございます！

Reply:
Python は全部「値渡し」だよ。「Python の引数は参照渡し」ってデマ書いてるサイト、めちゃくちゃ多いけど。

cf.
https://qiita.com/ur_kinsk/items/949dabe975bdc1affb82
https://qiita.com/raccy/items/d4c5e7995a8fc90109ee

このへん一度ちゃんと理解したら、検証しなくても色々一発で予想できるようになると思う。